### DISCUSSION TITLE
skip 1.1.8 export fails when fullscreencover is used

### DISCUSSION THREAD

**Discussion by utterances** [skiptools/skip#257] [Q&A] [answered]
for some reason when I use .fullscreencover it builds and runs fine in Xcode, and android studio, but fails on skip export during building... looks like it complains fullscreencover is not available for macOS. attached a example project and log, here's the relevant portion:

`'fullScreenCover(isPresented:onDismiss:content:)' is unavailable in macOS`

[skiptest.zip](https://github.com/user-attachments/files/17316175/skiptest.zip)
[skip-export-2024-10-09T17:38:14Z.txt](https://github.com/user-attachments/files/17316269/skip-export-2024-10-09T17.38.14Z.txt)

any ideas how to resolve? or do I have to avoid using fullscreencover? documentation does say it's fully supported in skip

**Comment by marcprux** [comment, marked as answer]
The problem is that `skip export` requires that the project be buildable for macOS. This is something we hope to fix in the future, but in the meantime you can work around it by guarding any iOS-specific API calls (like `fullScreenCover`) in an `#if !os(macOS)` block, like:

```swift
    public var body: some View {
        ContentView()
            #if !os(macOS)
            .fullScreenCover(isPresented:onDismiss:content:)
            #endif
    }
```

**Reply by utterances** [reply]
ah good to know... however when I tried this, the app will build and run, but on android the #if conditioned properties basically is not built, replaceable using the test project above. the export seems to complete without error this time but it's essentially disabling the fullscreencover code also for android build?

**Reply by marcprux** [reply]
What if you change it to `#if SKIP || os(iOS)`?
